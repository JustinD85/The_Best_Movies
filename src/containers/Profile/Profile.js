import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { setUser, setMovies, setPopup } from '../../actions'
import { getMovies } from '../../api/'
import './Profile.scss'

const Profile = (props) => {

  const logout = async () => {
    props.setUser(null)
    props.setMovies(await getMovies('now_playing'))
  }


  return (
    <div className='Profile' >
      <p>Profile</p>
      {!props.user ?
        <Link to='/login' onClick={() => props.setPopup(true)} >Login</Link>
        :
        <button onClick={() => logout()}>Logout</button>}
    </div>
  )
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  setUser: (user) => dispatch(setUser(user)),
  setMovies: (movies) => dispatch(setMovies(movies)),
  setPopup: (bool) => dispatch(setPopup(bool)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Profile);