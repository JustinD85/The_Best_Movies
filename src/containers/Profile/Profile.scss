.Profile {
  display: flex;
  justify-content: flex-end;
  padding: 2% 4% 0% 0%;
  font-weight: 600;

  .login-link {
    letter-spacing: 1px;
    color: #55E4F5;
    text-decoration: none;
    border-bottom: 2px solid transparent;
    padding-bottom: 8px;

    &:hover {
      border-bottom: 2px solid #55E4F5;
      cursor: pointer;
    }
  }
  
  .user-name {
    color:#ECEBEE;
    font-size: 17px;
    margin-right: 14px;
  }

  .logout-btn {
    letter-spacing: 1px;
    background-color: transparent;
    border: none;
    border-bottom: 2px solid transparent;
    color: #55E4F5;
    font-weight: bold;
    padding-bottom: 8px;

    &:hover {
      border-bottom: 2px solid #55E4F5;
      cursor: pointer;
    }
  }
}
