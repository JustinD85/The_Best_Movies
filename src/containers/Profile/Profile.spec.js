import React from 'react'
import { shallow } from 'enzyme'
import { Profile } from './Profile'

it('should pass a test', () => {
  let wrapper = shallow(<Profile user={{ name: 'J', id: 27 }} />)
  expect(wrapper).toMatchSnapshot()
})