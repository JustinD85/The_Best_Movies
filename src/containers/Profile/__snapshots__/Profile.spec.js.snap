// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass a test 1`] = `
<Fragment>
  <div
    className="Profile"
  >
    <p
      className="user-name"
    >
      J
    </p>
    <button
      className="logout-btn"
      onClick={[Function]}
    >
      SIGN OUT
    </button>
  </div>
</Fragment>
`;
